src/aosm/azext_aosm/build_processors/nfd_processor.py

```diff
@@ -96,10 +98,26 @@ def get_artifact_details(
             )
             and self.input_artifact.network_function_definition.properties.network_function_template
         ):
-            params = {
-                "nfvi_type":
-                self.input_artifact.network_function_definition.properties.network_function_template.nfvi_type
-            }
+            # Split for line-too-long linting errors
+            nf_type = self.input_artifact.network_function_definition.properties.network_function_type
+            nf_templates = self.input_artifact.network_function_definition.properties.network_function_template
+
+            if nf_type == "ContainerizedNetworkFunction":
+                nf_application_names = [nf_app.name for nf_app in nf_templates.network_function_applications]
+                params = {
+                    "nfvi_type":
+                    nf_templates.nfvi_type,
+                    "is_cnf": True,
+                    "nf_application_names": nf_application_names
+                }
+            elif nf_type == "VirtualNetworkFunction":
+                params = {
+                    "nfvi_type":
+                    nf_templates.nfvi_type,
+                    "is_cnf": False
+                }
+            else:
+                raise ResourceNotFoundError("The NFDV provided has no network function type.")
```

Comment:
if `network_function_type` doesn't exist then we are going to get an error on line 102 like we did when you tried running the test. What `else` here actually means is that we got a network function type but it does not equal to vnf or cnf. I would change the error message here to `invalid network function type %s`.

Reply:
Because this is a get from azure, the only options are cnf,vnf or unknown. so Iwill change the error to say that